modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredDecoder.scala

```diff
@@ -14,15 +16,30 @@ trait ConfiguredDecoder[A](using conf: Configuration) extends Decoder[A]:
   lazy val elemDefaults: Default[A]
   lazy val constructorNames: List[String] = elemLabels.map(conf.transformConstructorNames)
 
+  private lazy val decodersDict: Map[String, Decoder[?]] = {
+    def findDecoderDict(p: (String, Decoder[?])): List[(String, Decoder[?])] =
+      p._2 match {
+        case cd: ConfiguredDecoder[?] with SumOrProduct if cd.isSum =>
+          cd.constructorNames.zip(cd.elemDecoders).flatMap(findDecoderDict)
+        case _ => List(p)
+      }
+    constructorNames.zip(elemDecoders).flatMap(findDecoderDict).toMap
+  }
+
```

Comment:
I'm a little concerned that this doesn't handle recursive types such as:

```
sealed trait Tree; 
case class Branch(l: Tree, r: Tree) extends Tree; 
case object Leaf extends Tree;
```

But then I tried it locally and something else unrelated blows up first in the inlining, so this implementation is probably sufficient for now.

Reply:
Thanks! There was a recursive test stackoverflowed with my implementation before I changed to the code to stop traversing the tree at a `Product` decoder.  I am not sure if that change fixed it for the above scenario, but if we need to come back to this again feel free to ping me. Will be happy to continue improve it.

Reply:
Having a test for the code above would solve any possible concern :)